components/application-role-mgt/org.wso2.carbon.identity.application.role.mgt/src/main/java/org/wso2/carbon/identity/application/role/mgt/ApplicationRoleManagerImpl.java

```diff
@@ -0,0 +1,202 @@
+    public void updateApplicationRole(String applicationId, String roleId, String newName, List<String> addedScopes,
+                                      List<String> removedScopes) throws ApplicationRoleManagementException {
+
+        // TODO: Check authorized scopes for the app and filter out added permissions
+        applicationRoleMgtDAO.updateApplicationRole(roleId, newName, addedScopes, removedScopes, getTenantDomain());
+    }
+
+    @Override
+    public ApplicationRole getApplicationRoleById(String roleId) throws ApplicationRoleManagementException {
+
+        return applicationRoleMgtDAO.getApplicationRoleById(roleId, getTenantDomain());
+    }
+
+    @Override
+    public List<ApplicationRole> getApplicationRoles(String applicationId) throws ApplicationRoleManagementException {
+
+        return applicationRoleMgtDAO.getApplicationRoles(applicationId);
+    }
+
+    @Override
+    public void deleteApplicationRole(String roleId) throws ApplicationRoleManagementException {
+
+        applicationRoleMgtDAO.deleteApplicationRole(roleId, getTenantDomain());
+    }
+
+    @Override
+    public void updateApplicationRoleAssignedUsers(String roleId, List<String> addedUsers, List<String> removedUsers)
+            throws ApplicationRoleManagementException {
+
+        validateAppRoleId(roleId);
+        removeCommonValues(addedUsers, removedUsers);
+        applicationRoleMgtDAO.updateApplicationRoleAssignedUsers(roleId, addedUsers, removedUsers, getTenantDomain());
+    }
+
+    @Override
+    public ApplicationRole getApplicationRoleAssignedUsers(String roleId)
+            throws ApplicationRoleManagementException {
+
+        validateAppRoleId(roleId);
+        return applicationRoleMgtDAO.getApplicationRoleAssignedUsers(roleId, getTenantDomain());
+    }
+
+    @Override
+    public void updateApplicationRoleAssignedGroups(String roleId, String idpId, List<String> addedGroups,
+                                                    List<String> removedGroups)
+            throws ApplicationRoleManagementException {
+
+        validateAppRoleId(roleId);
+        try {
+            IdentityProvider identityProvider;
+            if (LOCAL_IDP.equals(idpId)) {
+
+                identityProvider = ApplicationRoleMgtServiceComponentHolder.getInstance()
+                        .getIdentityProviderManager().getResidentIdP(getTenantDomain());
+            } else {
+                identityProvider = ApplicationRoleMgtServiceComponentHolder.getInstance()
+                        .getIdentityProviderManager().getIdPByResourceId(idpId, getTenantDomain(), true);
+            }
+            removeCommonValues(addedGroups, removedGroups);
+            applicationRoleMgtDAO.updateApplicationRoleAssignedGroups(roleId, identityProvider,
+                    addedGroups, removedGroups, getTenantDomain());
+        }   catch (IdentityProviderManagementException e) {
+        throw new ApplicationRoleManagementException("Error while retrieving idp",
```

Comment:
Dont we need to send an error code from the manager level?

Reply:
move resolving idp to utils